src/main/java/org/jenkinsci/plugins/sma/SMARunner.java

```diff
@@ -58,16 +57,17 @@ public SMARunner(EnvVars jobVariables, String prTargetBranch, SMAJenkinsCIOrgSet
         }
         // Configure using pull request logic
         if (!prTargetBranch.isEmpty() && !shaOverride) {
-            deployAll = false;
-            git = new SMAGit(pathToWorkspace, currentCommit, prTargetBranch, SMAGit.Mode.PRB);
-            previousCommit = git.getPrevCommit();
+//            deployAll = false;
```

Comment:
I am sorry, but I don't agree with you. Maybe I am missing something, so please correct me if I am wrong! I think that without the `deployAll = false` it will go fine as `deployAll` is set to `false` by default and will only be set to `true` in case no `previousCommit` could be set and in the else clause line 47 is executed. What basically mean that only the first run will deploy all, or if `SMA_DEPLOY_ALL_METADATA` is set to `true`. So no need to set `deployAll = false` on line 60. On the contrary, setting it to `false` on line 60 will render the `SMA_DEPLOY_ALL_METADATA` environment variable useless in case a `prTargetBranch` is set, because it will always be set to `false`. That doesn't look intuitive to me. Hope this makes it more clear to you what I mean. Thanks.